usdt-impl/src/record.rs

```diff
@@ -169,16 +176,53 @@ fn limit_string_length<S: AsRef<str>>(s: S, limit: usize) -> String {
     s[..limit].to_string()
 }
 
-// Process a single record from the custom linker section.
-fn process_rec(providers: &mut BTreeMap<String, Provider>, rec: &[u8]) -> Result<(), crate::Error> {
-    // Skip over the length which was already read.
-    let mut data = &rec[4..];
+// Return the probe record version, atomically updating it if the probe record will be handled.
+fn read_and_update_record_version(data: &[u8]) -> Result<u8, crate::Error> {
+    // First check if we'll be handling this record at all. We support any version number less than
+    // or equal to the crate's version.
+    if data[0] <= PROBE_REC_VERSION {
+        // Atomically exchange the record's version with our sentinel value, and return the
+        // record's version.
+        //
+        // NOTE: It's not easy to use types from `std::sync::atomic` here because we need to
+        // atomically exchange the data through a pointer. `AtomicU8::from_mut` might work, but
+        // that would require another feature flag pinning us to a nightly compiler.
+        let mut version = u8::MAX;
+        let mut existing_version = data.as_ptr();
+        #[allow(unused_assignments)]
+        unsafe {
+            asm!(
+                "lock xchg al, [{}]",
+                inout(reg) existing_version,
```

Comment:
is this `inout` or just `in`?

Reply:
Nope, nice catch. Fixed and removed the `mut` annotation in bc48dc9